File: storage/db-core/src/main/java/com/coffee/gu/menu/OptionRepositoryImpl.java
package com.coffee.gu.menu;

import com.coffee.gu.enums.EntityStatus;
import com.querydsl.jpa.impl.JPAQueryFactory;
import org.springframework.stereotype.Repository;

import java.util.List;

import static com.coffee.gu.menu.QOptionEntity.optionEntity;

@Repository
public class OptionRepositoryImpl implements OptionRepository{

    private final JPAQueryFactory queryFactory;

    public OptionRepositoryImpl(JPAQueryFactory queryFactory) {
        this.queryFactory = queryFactory;
    }

    @Override
    public List<Option> findAllByIdOptionGroupIdIn(List<Long> optionGroupIds) {
        return queryFactory.selectFrom(optionEntity)
                .where(
                        optionEntity.optionGroupId.in(optionGroupIds),
                        optionEntity.entityStatus.eq(EntityStatus.ACTIVE)
                )
                .fetch()
                .stream()
                .map(OptionEntity::toModel)
                .toList();
    }
}

File: storage/db-core/src/main/kotlin/com/coffee/gu/menu/OptionRepositoryImpl.kt
package com.coffee.gu.menu

import com.coffee.gu.enums.EntityStatus
import com.coffee.gu.menu.QOptionEntity.optionEntity
import com.querydsl.jpa.impl.JPAQueryFactory
import org.springframework.stereotype.Repository

@Repository
class OptionRepositoryImpl(
    private val queryFactory: JPAQueryFactory,
) : OptionRepository {

    override fun findAllByIdOptionGroupIdIn(optionGroupIds: List<Long>): List<Option> {
        return queryFactory.selectFrom(optionEntity)
            .where(
                optionEntity.optionGroupId.`in`(optionGroupIds),
                optionEntity.entityStatus.eq(EntityStatus.ACTIVE),
            )
            .fetch()
            .map { it.toModel() }
    }
}
